### ISSUE TITLE
Can't create composer preview versions

### ISSUE THREAD

**Issue by hlastraspfx** [hashicorp/terraform-provider-google#9201]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

Terraform v0.15.3
on darwin_amd64

### Affected Resource(s)

google_composer_environment

### Terraform Configuration Files

```tf
resource "google_composer_environment" "airflow" {
  provider = google-beta
  [...]
  software_config {
    image_version  = "composer-1.17.0-preview.0-airflow-2.0.1"
    [...]
  }
}
```

### Expected Behavior

Creation of the resource.

### Actual Behavior

Error:
```
Error: “config.0.software_config.0.image_version” (“composer-1.17.0-preview.0-airflow-2.0.1") doesn’t match regexp “composer-([0-9]+\\.[0-9]+\\.[0-9]+|latest)-airflow-([0-9]+\\.[0-9]+(\\.[0-9]+.*)?)”
```

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by hlastraspfx** [comment]
I have a solution ready:

```
diff --git a/google/resource_composer_environment.go b/google/resource_composer_environment.go
index fa00add3f..a3f5b4111 100644
--- a/google/resource_composer_environment.go
+++ b/google/resource_composer_environment.go
@@ -16,7 +16,7 @@ import (
 const (
        composerEnvironmentEnvVariablesRegexp          = "[a-zA-Z_][a-zA-Z0-9_]*."
        composerEnvironmentReservedAirflowEnvVarRegexp = "AIRFLOW__[A-Z0-9_]+__[A-Z0-9_]+"
-       composerEnvironmentVersionRegexp               = `composer-([0-9]+\.[0-9]+\.[0-9]+|latest)-airflow-([0-9]+\.[0-9]+(\.[0-9]+.*)?)`
+       composerEnvironmentVersionRegexp               = `composer-([0-9]+\.[0-9]+\.[0-9]+(-preview\.[0-9]+)?|latest)-airflow-([0-9]+\.[0-9]+(\.[0-9]+.*)?)`
 )
 
 var composerEnvironmentReservedEnvVar = map[string]struct{}{
diff --git a/google/resource_composer_environment_test.go b/google/resource_composer_environment_test.go
index eb3cc8ce0..5da8d22f0 100644
--- a/google/resource_composer_environment_test.go
+++ b/google/resource_composer_environment_test.go
@@ -40,6 +40,7 @@ func TestComposerImageVersionDiffSuppress(t *testing.T) {
                {"airflow equivalent", "composer-1.4.0-airflow-1.10.0", "composer-1.4.0-airflow-1.10", true},
                {"airflow different", "composer-1.4.0-airflow-1.10.0", "composer-1.4-airflow-1.9.0", false},
                {"airflow different composer latest", "composer-1.4.0-airflow-1.10.0", "composer-latest-airflow-1.9.0", false},
+               {"preview", "composer-1.17.0-preview.0-airflow-2.0.1", "composer-1.17.0-preview.0-airflow-2.0.1", true},
        }
 
        for _, tc := range cases {
```

**Comment by scdekov** [comment]
Is there any workaround that we can use until this is fixed?

**Comment by melinath** [comment]
It looks like this will be resolved by https://github.com/GoogleCloudPlatform/magic-modules/pull/4797

**Comment by melinath** [comment]
This is causing failures of the following tests:

- TestAccComposerEnvironment_update
- TestAccComposerEnvironment_withSoftwareConfig

**Comment by krishna4ldw** [comment]
Is this fixed in new terraform version or google provider version for terraform?

**Comment by rileykarson** [comment]
This'll be fixed in a provider version, likely `3.71.0` on June 6th.
